kadena/core/src/crypto/hash/sha512.rs

```diff
@@ -3,11 +3,11 @@
 
 use crate::crypto::error::CryptoError;
 use crate::crypto::hash::HashValue;
-use sha2::{Digest, Sha512Trunc256};
+use sha2::{Digest, Sha512_256};
 
 // Hash functions for Merkle tree nodes
 // cf. https://github.com/kadena-io/chainweb-node/wiki/Chainweb-Merkle-Tree#merke-log-trees
-pub type ChainwebHash = Sha512Trunc256;
+pub type ChainwebHash = Sha512_256;
```

Comment:
Is this change necessary?

Reply:
From my understanding, `Sha512Trunc256` is relevant for the [0.9](https://github.com/RustCrypto/hashes/blob/backports/digest-v0.9/sha2/src/lib.rs) version of `sha2` crate which is quite outdated, while our zkvm branch contains more recent version (10.0.8) with specifically [exposed types](https://github.com/argumentcomputer/RustCrypto-hashes/blob/zkvm/sha2/src/lib.rs) for hasher instantiation. So having that we (I believe) want to use both sha512 and blake2s sphinx precompiles integrated via "unified" dependency which is currently [zkvm](https://github.com/argumentcomputer/RustCrypto-hashes/tree/zkvm) branch

Reply:
I see, it's an upstream `sha2` type change, thanks!